--- ../src_base/minecraft_server/net/minecraft/src/CommandServerDeop.java
+++ ../src_work/minecraft_server/net/minecraft/src/CommandServerDeop.java
@@ -1,4 +1,4 @@
-package net.minecraft.src;
+/*package net.minecraft.src;
 
 import java.util.List;
 import net.minecraft.server.MinecraftServer;
@@ -10,9 +10,6 @@
         return "deop";
     }
 
-    /**
-     * Return the required permission level for this command.
-     */
     public int getRequiredPermissionLevel()
     {
         return 3;
@@ -36,11 +33,9 @@
         }
     }
 
-    /**
-     * Adds the strings available in this command to the given list of tab completion options.
-     */
     public List addTabCompletionOptions(ICommandSender par1ICommandSender, String[] par2ArrayOfStr)
     {
         return par2ArrayOfStr.length == 1 ? getListOfStringsFromIterableMatchingLastWord(par2ArrayOfStr, MinecraftServer.getServer().getConfigurationManager().getOps()) : null;
     }
 }
+*/