--- ../src_base/minecraft_server/net/minecraft/src/EntityPlayerMP.java
+++ ../src_work/minecraft_server/net/minecraft/src/EntityPlayerMP.java
@@ -9,6 +9,8 @@
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
+
+import jk_5.nailed.NailedEventFactory;
 import net.minecraft.server.MinecraftServer;
 
 public class EntityPlayerMP extends EntityPlayer implements ICrafting {
@@ -39,16 +41,10 @@
       par4ItemInWorldManager.thisPlayerMP = this;
       this.theItemInWorldManager = par4ItemInWorldManager;
       this.renderDistance = par1MinecraftServer.getConfigurationManager().getViewDistance();
-      ChunkCoordinates var5 = par2World.getSpawnPoint();
+      ChunkCoordinates var5 = NailedEventFactory.getSpawnPoint(this);
       int var6 = var5.posX;
       int var7 = var5.posZ;
       int var8 = var5.posY;
-      if(!par2World.provider.hasNoSky && par2World.getWorldInfo().getGameType() != EnumGameType.ADVENTURE) {
-         int var9 = Math.max(5, par1MinecraftServer.getSpawnProtectionSize() - 6);
-         var6 += this.rand.nextInt(var9 * 2) - var9;
-         var7 += this.rand.nextInt(var9 * 2) - var9;
-         var8 = par2World.getTopSolidOrLiquidBlock(var6, var7);
-      }
 
       this.mcServer = par1MinecraftServer;
       this.stepHeight = 0.0F;
@@ -244,7 +240,7 @@
    }
 
    public boolean func_96122_a(EntityPlayer par1EntityPlayer) {
-      return !this.mcServer.isPVPEnabled()?false:super.func_96122_a(par1EntityPlayer);
+       return !NailedEventFactory.canPlayerAttackPlayer(this, par1EntityPlayer) ? false:(!this.mcServer.isPVPEnabled()?false:super.func_96122_a(par1EntityPlayer));
    }
 
    public void travelToTheEnd(int par1) {
