--- ../src_base/minecraft_server/net/minecraft/src/EntityItem.java
+++ ../src_work/minecraft_server/net/minecraft/src/EntityItem.java
@@ -1,4 +1,6 @@
 package net.minecraft.src;
+
+import jk_5.nailed.NailedEventFactory;
 
 import java.util.Iterator;
 
@@ -282,7 +284,7 @@
      */
     public void onCollideWithPlayer(EntityPlayer par1EntityPlayer)
     {
-        if (!this.worldObj.isRemote)
+        if (!this.worldObj.isRemote && NailedEventFactory.canPlayerPickup(par1EntityPlayer))
         {
             ItemStack var2 = this.getEntityItem();
             int var3 = var2.stackSize;
