--- ../src_base/minecraft_server/net/minecraft/src/EntityArrow.java
+++ ../src_work/minecraft_server/net/minecraft/src/EntityArrow.java
@@ -1,4 +1,6 @@
 package net.minecraft.src;
+
+import jk_5.nailed.NailedEventFactory;
 
 import java.util.List;
 
@@ -344,6 +346,7 @@
 
    public void onCollideWithPlayer(EntityPlayer par1EntityPlayer) {
       if(!this.worldObj.isRemote && this.inGround && this.arrowShake <= 0) {
+         if(!NailedEventFactory.canPlayerPickup(par1EntityPlayer)) return;
          boolean var2 = this.canBePickedUp == 1 || this.canBePickedUp == 2 && par1EntityPlayer.capabilities.isCreativeMode;
          if(this.canBePickedUp == 1 && !par1EntityPlayer.inventory.addItemStackToInventory(new ItemStack(Item.arrow, 1))) {
             var2 = false;
