--- ../src_base/minecraft_server/net/minecraft/src/NetServerHandler.java
+++ ../src_work/minecraft_server/net/minecraft/src/NetServerHandler.java
@@ -6,6 +6,8 @@
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.Random;
+
+import jk_5.nailed.NailedEventFactory;
 import net.minecraft.server.MinecraftServer;
 
 public class NetServerHandler extends NetHandler
@@ -102,7 +104,6 @@
             this.playerEntity.mountEntityAndWakeUp();
             this.sendPacket(new Packet255KickDisconnect(par1Str));
             this.netManager.serverShutdown();
-            this.mcServer.getConfigurationManager().sendChatMsg(ChatMessageComponent.createFromTranslationWithSubstitutions("multiplayer.player.left", new Object[] {this.playerEntity.getTranslatedEntityName()}).setColor(EnumChatFormatting.YELLOW));
             this.mcServer.getConfigurationManager().playerLoggedOut(this.playerEntity);
             this.connectionClosed = true;
         }
@@ -524,7 +525,6 @@
     public void handleErrorMessage(String par1Str, Object[] par2ArrayOfObj)
     {
         this.mcServer.getLogAgent().logInfo(this.playerEntity.getCommandSenderName() + " lost connection: " + par1Str);
-        this.mcServer.getConfigurationManager().sendChatMsg(ChatMessageComponent.createFromTranslationWithSubstitutions("multiplayer.player.left", new Object[] {this.playerEntity.getTranslatedEntityName()}).setColor(EnumChatFormatting.YELLOW));
         this.mcServer.getConfigurationManager().playerLoggedOut(this.playerEntity);
         this.connectionClosed = true;
 
@@ -633,8 +633,7 @@
                         return;
                     }
 
-                    ChatMessageComponent var4 = ChatMessageComponent.createFromTranslationWithSubstitutions("chat.type.text", new Object[] {this.playerEntity.getTranslatedEntityName(), var2});
-                    this.mcServer.getConfigurationManager().func_110459_a(var4, false);
+                    NailedEventFactory.onPlayerChat(this.playerEntity, var2);
                 }
 
                 this.chatSpamThresholdCount += 20;
