--- ../src_base/minecraft_server/net/minecraft/src/ServerConfigurationManager.java
+++ ../src_work/minecraft_server/net/minecraft/src/ServerConfigurationManager.java
@@ -13,6 +13,9 @@
 import java.util.Set;
 import java.util.Map.Entry;
 
+import jk_5.nailed.Nailed;
+import jk_5.nailed.event.player.PlayerJoinServerEvent;
+import jk_5.nailed.event.player.PlayerLeaveServerEvent;
 import net.minecraft.server.MinecraftServer;
 
 public abstract class ServerConfigurationManager
@@ -86,7 +89,7 @@
         var7.sendPacket(new Packet16BlockItemSwitch(par2EntityPlayerMP.inventory.currentItem));
         this.func_96456_a((ServerScoreboard)var5.getScoreboard(), par2EntityPlayerMP);
         this.updateTimeAndWeatherForPlayer(par2EntityPlayerMP, var5);
-        this.sendChatMsg(ChatMessageComponent.func_111082_b("multiplayer.player.joined", new Object[] {par2EntityPlayerMP.getTranslatedEntityName()}).func_111059_a(EnumChatFormatting.YELLOW));
+        //NAILED DISABLE this.sendChatMsg(ChatMessageComponent.func_111082_b("multiplayer.player.joined", new Object[] {par2EntityPlayerMP.getTranslatedEntityName()}).func_111059_a(EnumChatFormatting.YELLOW));
         this.playerLoggedIn(par2EntityPlayerMP);
         var7.setPlayerLocation(par2EntityPlayerMP.posX, par2EntityPlayerMP.posY, par2EntityPlayerMP.posZ, par2EntityPlayerMP.rotationYaw, par2EntityPlayerMP.rotationPitch);
         this.mcServer.getNetworkThread().addPlayer(var7);
@@ -213,6 +216,10 @@
      */
     public void playerLoggedIn(EntityPlayerMP par1EntityPlayerMP)
     {
+        //NAILED START
+        Nailed.eventBus.post(new PlayerJoinServerEvent(par1EntityPlayerMP));
+        //NAILED END
+
         this.sendPacketToAllPlayers(new Packet201PlayerInfo(par1EntityPlayerMP.getCommandSenderName(), true, 1000));
         this.playerEntityList.add(par1EntityPlayerMP);
         WorldServer var2 = this.mcServer.worldServerForDimension(par1EntityPlayerMP.dimension);
