--- ../src_base/minecraft_server/net/minecraft/src/ServerCommandManager.java
+++ ../src_work/minecraft_server/net/minecraft/src/ServerCommandManager.java
@@ -1,10 +1,13 @@
 package net.minecraft.src;
 
 import java.util.Iterator;
+
+import jk_5.nailed.Nailed;
 import net.minecraft.server.MinecraftServer;
 
 public class ServerCommandManager extends CommandHandler implements IAdminCommand {
    public ServerCommandManager() {
+      Nailed.registerCommands(this);
       this.registerCommand(new CommandTime());
       this.registerCommand(new CommandGameMode());
       this.registerCommand(new CommandDifficulty());
