--- ../src_base/minecraft_server/net/minecraft/src/TileEntityCommandBlock.java
+++ ../src_work/minecraft_server/net/minecraft/src/TileEntityCommandBlock.java
@@ -1,5 +1,6 @@
 package net.minecraft.src;
 
+import jk_5.nailed.map.stats.StatManager;
 import net.minecraft.server.MinecraftServer;
 
 public class TileEntityCommandBlock extends TileEntity implements ICommandSender
@@ -60,6 +61,7 @@
     public void setCommandSenderName(String par1Str)
     {
         this.commandSenderName = par1Str;
+        StatManager.updateCommandBlock(this);
     }
 
     public void sendChatToPlayer(ChatMessageComponent par1ChatMessageComponent) {}
@@ -95,6 +97,7 @@
         if (par1NBTTagCompound.hasKey("CustomName"))
         {
             this.commandSenderName = par1NBTTagCompound.getString("CustomName");
+            StatManager.updateCommandBlock(this);
         }
     }
 
