--- ../src_base/minecraft_server/net/minecraft/src/WorldProvider.java
+++ ../src_work/minecraft_server/net/minecraft/src/WorldProvider.java
@@ -1,4 +1,6 @@
 package net.minecraft.src;
+
+import jk_5.nailed.map.NailedWorldGen;
 
 public abstract class WorldProvider
 {
@@ -71,6 +73,7 @@
         {
             this.worldChunkMgr = new WorldChunkManager(this.worldObj);
         }
+        this.worldChunkMgr = NailedWorldGen.getWorldChunkManager(this.worldObj);
     }
 
     /**
@@ -78,7 +81,7 @@
      */
     public IChunkProvider createChunkGenerator()
     {
-        return (IChunkProvider)(this.terrainType == WorldType.FLAT ? new ChunkProviderFlat(this.worldObj, this.worldObj.getSeed(), this.worldObj.getWorldInfo().isMapFeaturesEnabled(), this.field_82913_c) : new ChunkProviderGenerate(this.worldObj, this.worldObj.getSeed(), this.worldObj.getWorldInfo().isMapFeaturesEnabled()));
+        return NailedWorldGen.getChunkProvider(this.worldObj);
     }
 
     /**
