--- /net/minecraft/entity/player/EntityPlayerMP.java
+++ /net/minecraft/entity/player/EntityPlayerMP.java
@@ -9,6 +9,8 @@
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
+
+import jk_5.nailed.NailedEventFactory;
 import net.minecraft.crash.CrashReport;
 import net.minecraft.crash.CrashReportCategory;
 import net.minecraft.entity.Entity;
@@ -163,18 +165,11 @@
         par4ItemInWorldManager.thisPlayerMP = this;
         this.theItemInWorldManager = par4ItemInWorldManager;
         this.renderDistance = par1MinecraftServer.getConfigurationManager().getViewDistance();
-        ChunkCoordinates var5 = par2World.getSpawnPoint();
+        ChunkCoordinates var5 = NailedEventFactory.getSpawnPoint(this);
         int var6 = var5.posX;
         int var7 = var5.posZ;
         int var8 = var5.posY;
 
-        if (!par2World.provider.hasNoSky && par2World.getWorldInfo().getGameType() != EnumGameType.ADVENTURE) {
-            int var9 = Math.max(5, par1MinecraftServer.getSpawnProtectionSize() - 6);
-            var6 += this.rand.nextInt(var9 * 2) - var9;
-            var7 += this.rand.nextInt(var9 * 2) - var9;
-            var8 = par2World.getTopSolidOrLiquidBlock(var6, var7);
-        }
-
         this.mcServer = par1MinecraftServer;
         this.stepHeight = 0.0F;
         this.yOffset = 0.0F;
@@ -404,7 +399,7 @@
     }
 
     public boolean canAttackPlayer(EntityPlayer par1EntityPlayer) {
-        return !this.mcServer.isPVPEnabled() ? false : super.canAttackPlayer(par1EntityPlayer);
+        return NailedEventFactory.canPlayerAttackPlayer(this, par1EntityPlayer);
     }
 
     /**
