--- /net/minecraft/entity/item/EntityXPOrb.java
+++ /net/minecraft/entity/item/EntityXPOrb.java
@@ -1,5 +1,6 @@
 package net.minecraft.entity.item;
 
+import jk_5.nailed.NailedEventFactory;
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.entity.Entity;
@@ -190,7 +191,7 @@
      * Called by a player entity when they collide with an entity
      */
     public void onCollideWithPlayer(EntityPlayer par1EntityPlayer) {
-        if (!this.worldObj.isRemote) {
+        if (!this.worldObj.isRemote && NailedEventFactory.canPlayerPickup(par1EntityPlayer)) {
             if (this.field_70532_c == 0 && par1EntityPlayer.xpCooldown == 0) {
                 par1EntityPlayer.xpCooldown = 2;
                 this.playSound("random.orb", 0.1F, 0.5F * ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.7F + 1.8F));
