--- /net/minecraft/command/CommandServerOp.java
+++ /net/minecraft/command/CommandServerOp.java
@@ -1,4 +1,4 @@
-package net.minecraft.command;
+/*package net.minecraft.command;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -9,9 +9,6 @@
         return "op";
     }
 
-    /**
-     * Return the required permission level for this command.
-     */
     public int getRequiredPermissionLevel() {
         return 3;
     }
@@ -29,9 +26,6 @@
         }
     }
 
-    /**
-     * Adds the strings available in this command to the given list of tab completion options.
-     */
     public List addTabCompletionOptions(ICommandSender par1ICommandSender, String[] par2ArrayOfStr) {
         if (par2ArrayOfStr.length == 1) {
             String var3 = par2ArrayOfStr[par2ArrayOfStr.length - 1];
@@ -52,4 +46,4 @@
             return null;
         }
     }
-}
+}*/
